### PR TITLE
feat: add ai chatbot to member profile 🤖

### LINKED ISSUE #541: Add AI Chatbot in Member Profile 🤖
**Issue Reserved for Bloomberg Mentorship Program**

### Description

Right now, members can ask a question to the AI Chatbot by asking the ColorStack bot within the Slack. However, we’d like for members also be able to communicate with the bot on the Member Profile. We want to create a new tab on the member profile where members can see a simple interface that allows them to write up a question, submit the question, and then get a response from the ColorStack bot. 

### Acceptance Criteria

- Create a new tab in the member directory
- Create a simple interface on the new page that has (1) a textbox for users to write their question,  (2) a button for users to submit their question, (3) a box for users to see the response from the from the chatbot, (4) a button to clear the response textbox and the question textbox to ask a new question
- Create a new function that takes a question, uses [IsQuestion](https://github.com/colorstackorg/oyster/blob/57032d5d9bfe81d49935aa6e2d9d213c0f6806e6/packages/core/src/modules/slack/slack.ts#L427-L462) to verify if it is a valid question, and then uses [getAnswerFromSlackHistory](https://github.com/colorstackorg/oyster/blob/57032d5d9bfe81d49935aa6e2d9d213c0f6806e6/packages/core/src/modules/slack/slack.ts#L464-L478) and [addThreadReferences](https://github.com/colorstackorg/oyster/blob/57032d5d9bfe81d49935aa6e2d9d213c0f6806e6/packages/core/src/modules/slack/slack.ts#L593-L608) to return a response
- When a user presses the submit button with a non-empty message, this should trigger the new function and return a response that appears in the response textbox.
- When the button to ask a new question is pressed, the question textbox and the response textbox should be cleared.

### Additional Context

Look at the code for how we implement the chatbot service for the slack with the function [answerChatbotQuestion](https://github.com/colorstackorg/oyster/blob/57032d5d9bfe81d49935aa6e2d9d213c0f6806e6/packages/core/src/modules/slack/slack.ts#L48-L138). More generally, the file [slack.ts](https://github.com/colorstackorg/oyster/blob/main/packages/core/src/modules/slack/slack.ts) provides a lot of functions that are relevant to understanding and implementing this issue.

### PR DESCRIPTION
**Pull request by Boris713**
## Description ✏️

Closes #541 

Describe what this PR does.

- Adds new tab/page for AI chat bot in member profiles
- Allows users to ask questions and have the ai answer them based off info from slack
Screenshot 2024-10-26 at 7 33 59 PM

## Type of Change 🐞

- [x] Feature - A non-breaking change which adds functionality.
- [ ] Fix - A non-breaking change which fixes an issue.
- [ ] Refactor - A change that neither fixes a bug nor adds a feature.
- [ ] Documentation - A change only to in-code or markdown documentation.
- [ ] Tests - A change that adds missing unit/integration tests.
- [ ] Chore - A change that is likely none of the above.

## Checklist ✅

- [x] I have done a self-review of my code.
- [x] I have manually tested my code (if applicable).
- [ ] I have added/updated any relevant documentation (if applicable).

## Areas of Interest 👀
- I want to ensure exporting of AnswerMemberProfileQuestion is done correctly. specifically using member-profile.server.ts
- Unable to test directly due to not having API credits. Although it should work, would be good to test.
- Order of icons in sidebar

### CODE REVIEW

File: apps/member-profile/app/routes/_profile.ai-chat-bot.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                </div>
+                <div className="rounded-lg border border-[#348E87] bg-white p-4 shadow">
+                  <p className="text-base text-gray-800">{response}</p>
+                </div>
+              </div>
+            ) : (
+              <div className="flex h-full items-center justify-center">
+                <p className="text-center text-gray-500">
+                  {
+                    'Ask me anything and Ill search the ColorStack Slack for you!'
```
Comment (tomas-salgado) [inline apps/member-profile/app/routes/_profile.ai-chat-bot.tsx:70]: typo: 'I'll' instead of Ill

**Review by tomas-salgado — CHANGES_REQUESTED** [review]
Great work @Boris713! The UI looks great. I wasn't able to test it fully because I don't have access to all of the embeddings for the Slack messages, but I'll ping Rami and he'll be able to help with this. 

Just left a couple of minor comments, after that this should be good for a final review

**Comment by Boris713** [comment]
Fixed! can I get a LGTM?

**Review by tomas-salgado — APPROVED** [review]
LGTM!

**Review by ramiAbdou — APPROVED** [review]
Great work, @Boris713! 🚀

I made some updates and wanted to summarize them here:
- Since we're going to be rendering the response in the UI and not Slack, we needed to do some additional formatting of the Slack answer such that we can make the UI dynamic. For example, in order to render the "reference links" we need to parse the string into multiple "parts", when of them being a `reference` which our UI can now then work off of.
- The UI is updated such that it supports the above, and also it has a couple suggested prompts that the user can work off. We also now have a custom loading state since the response can take many seconds to come up.
- We'll cache the last question/answer that was asked for an hour so that the member doesn't lose it if they navigate away.

I'm going to follow up with some additional formatting for the sources that it retrieved the answer to as well. Take a look!

Screenshot 2024-11-07 at 4 18 29 PM

Screenshot 2024-11-07 at 4 16 22 PM

**Pull request merged**